provider/core/src/datagen/payload.rs

```diff
@@ -0,0 +1,147 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::dynutil::UpcastDataPayload;
+use crate::prelude::*;
+use crate::yoke::*;
+use alloc::boxed::Box;
+use crabbake::{Bakeable, CrateEnv, TokenStream};
+
+trait ExportableYoke {
```

Comment:
issue: a small concern i have here is that this doesn't really let folks extend this with types that only support serialization or baking, perhaps when working on new backends that can't be baked. We could use the ZeroFrom trick or make the impl panicky in those cases though. Idk, thoughts?

Reply:
In general we want all data structs to support baking and serde. We can mostly derive Bake anyway, and if that's not possible it could panic or return some constant. I don't think we want to ever use the ZF hack again, because that means we're designing a new data struct to be non-zero-copy